R/map_peptides_on_structure.R

```diff
@@ -0,0 +1,342 @@
+#' Maps peptides onto a PDB structure
+#'
+#' Peptides are mapped onto PDB structures based on their positions. This is accomplished by replacing the B-factor information in the structure file with values that allow
+#' highlighting peptides when the structure is coloured by B-factor. In addition to only highlighting peptides, a continuous variable such as fold changes associated with peptides
+#' can be mapped onto the structure as a colour gradient.
+#'
+#' @param peptide_data a data frame containing at least the input columns to this function. If a structure file is provided in the \code{structure_file} argument,
+#' this data frame should only contain peptide and structure information for the provided structure since otherwise the information cannot be mapped correctly to the file due
+#' to the missing PDB identifier.
+#' @param uniprot_id a character column in the \code{peptide_data} data frame that contains UniProt identifiers for a corresponding peptide.
+#' @param pdb_id a character column in the \code{peptide_data} data frame that contains PDB identifiers for structures in which a corresponding peptide is found. This
+#' column is not required if a structure file is provided in the \code{structure_file} argument.
+#' @param chain a character column in the \code{peptide_data} data frame that contains the name of the chain from the PDB structure in which the peptide is found. **Important:**
+#' please provide the author defined chain definitions for both ".cif" and ".pdb" files. When the output of the \code{find_peptide_in_pdb} function
+#' is used as the input for this function, this corresponds to the \code{auth_chain} column.
+#' @param start_in_pdb a numeric column in the \code{peptide_data} data frame that contains start positions for peptides in the corresponding PDB structure. This information
+#' can be obtained from the \code{find_peptide_in_pdb} function. The corresponding column in the output is called \code{peptide_start_pdb}.
+#' @param end_in_pdb a numeric column in the \code{peptide_data} data frame that contains end positions for peptides in the corresponding PDB structure. This information
+#' can be obtained from the \code{find_peptide_in_pdb} function. The corresponding column in the output is called \code{peptide_end_pdb}.
+#' @param map_value a numeric column in the \code{peptide_data} data frame that contains a value associated with each peptide. This value will be displayed as
+#' a colour gradient when mapped onto the structure. The value can for example be the fold change, p-value or score associated with each peptide. If peptides should
+#' be displayed with just one colour, the value in this column should be the same for every peptide. For the mapping, values are scaled between 50 and 100. Regions in the structure
+#' that do not map any peptide receive a value of 0. If an amino acid position is associated with multiple scores, e.g. from different peptides, the
```

Comment:
instead of "score" I would use "map_value" here (mapping value does not necessarily have to be a score)